### PR TITLE
example: add challenge number

### PR DESCRIPTION
**Pull request by mieszko4**
Modifies Bunny Twirl to show how to use `Rune.getChallengeNumber()`:
* for game map randomisation using challenge number as a seed to pseudorandom function
* for getting game map from ad fixed list using modulo operator.

### CODE REVIEW

File: examples/bunny-twirl/seedrandom.min.js [hunk 1]
```diff
@@ -0,0 +1 @@
... n;return s&&(n=s.randomBytes)?n=n(l):(n=new Uint8Array(l),(f.crypto||f.msCrypto).getRandomValues(n)),S(n)}catch(n){var t=f.navigator,r=t&&t.plugins;return[+new Date,f,r,f.screen,S(a)]}}():n,3),o),u=new m(o);return e.int32=function(){return 0|u.g(4)},e.quick=function(){return u.g(4)/4294967296},e.double=e,j(S(u.S),a),(t.pass||r||function(n,t,r,e){return e&&(e.S&&v(e,u),n.state=function(){return v(u,{})}),r?(c[p]=n,t):n})(e,i,"global"in t?t.global:this==c,t.state)}function m(n){var t,r=n.length,u=this,e=0,o=u.i=u.j=0,i=u.S=[];for(r||(n=[r++]);e<l;)i[e]=e++;for(e=0;e<l;e++)i[e]=i[o=h&o+n[e%r]+(t=i[e])],i[o]=t;(u.g=function(n){for(var t,r=0,e=u.i,o=u.j,i=u.S;n--;)t=i[e=h&e+1],r=r*l+i[h&(i[e]=i[o=h&o+t])+(i[o]=t)];return u.i=e,u.j=o,r})(l)}function v(n,t){return t.i=n.i,t.j=n.j,t.S=n.S.slice(),t}function j(n,t){for(var r,e=n+"",o=0;o<e.length;)t[h&o]=h&(r^=19*t[h&o])+e.charCodeAt(o++);return S(t)}function S(n){return String.fromCharCode.apply(0,n)}if(j(c.random(),a),"object"==typeof module&&module.exports){module.exports=n;try{s=require("crypto")}catch(n){}}else"function"==typeof define&&define.amd?define(function(){return n}):c["seed"+p]=n}("undefined"!=typeof self?self:this,[],Math);
```
Comment (mattbretl) [inline examples/bunny-twirl/seedrandom.min.js:14]: Let's add a link to the source and include the license text.

File: examples/bunny-twirl/index.js [hunk 2]
```diff
@@ -56,13 +56,26 @@ container.addChild(text)
...
 let isPlaying = false
 
+// Generate map
+const challengeNumber = Rune.getChallengeNumber();
+// const nextChallengeNumber = challengeNumber + 1;
+
+// 1. Game map randomization
+const random = new Math.seedrandom(challengeNumber)
+const pseudoRandomNumber = random()
+const speed = (pseudoRandomNumber / 100) * 3 + 0.005
```
Comment (mattbretl) [inline examples/bunny-twirl/index.js]: Is the `+ 0.005` to ensure speed is at least 0.005? If so, using `Math.min` instead would make that clear.

File: examples/bunny-twirl/index.js [hunk 3]
```diff
@@ -56,13 +56,26 @@ container.addChild(text)
 let isAnimating = true
 let isPlaying = false
 
+// Generate map
+const challengeNumber = Rune.getChallengeNumber();
+// const nextChallengeNumber = challengeNumber + 1;
+
+// 1. Game map randomization
```
Comment (mattbretl) [inline examples/bunny-twirl/index.js:63]: I realize this is aligned with the README section title, but it seems odd in the context of this game. Should we add something like `// Derive rotation speed (floating-point number between 0.005 and 0.03) from Rune challenge number` to explain what we're actually doing?

File: examples/bunny-twirl/index.js [hunk 4]
```diff
@@ -56,13 +56,26 @@ container.addChild(text)
...
+const random = new Math.seedrandom(challengeNumber)
+const pseudoRandomNumber = random()
+const speed = (pseudoRandomNumber / 100) * 3 + 0.005
+
+// 2. Fixed list of game maps
```
Comment (mattbretl) [inline examples/bunny-twirl/index.js:71]: Similar to my other comment, should we add something like `// Derive direction (-1 or 1) from Rune challenge number`?

File: examples/bunny-twirl/index.js (hunk 2 above)
Comment (mieszko4) [reply examples/bunny-twirl/index.js]: Let me use the standard way as per https://stackoverflow.com/questions/4959975/generate-random-number-between-two-numbers-in-javascript to make it more clear.

Comment (mieszko4) [reply examples/bunny-twirl/index.js]: Fixed in https://github.com/rune/rune-games-sdk/pull/24/commits/d3f340842c4b95f6588463bf94299bc7357406b7

File: examples/bunny-twirl/index.js (hunk 3 above)
Comment (mieszko4) [reply examples/bunny-twirl/index.js:63]: Fixed in https://github.com/rune/rune-games-sdk/pull/24/commits/75d27bf8f857bf99a576dc5fd7abcb06909a0d51

File: examples/bunny-twirl/index.js (hunk 4 above)
Comment (mieszko4) [reply examples/bunny-twirl/index.js:71]: Fixed in https://github.com/rune/rune-games-sdk/pull/24/commits/75d27bf8f857bf99a576dc5fd7abcb06909a0d51

File: examples/bunny-twirl/seedrandom.min.js (hunk 1 above)
Comment (mieszko4) [reply examples/bunny-twirl/seedrandom.min.js:14]: Fixed in https://github.com/rune/rune-games-sdk/pull/24/commits/f2cf707fdd3cabbd988bd550bcfa7769e55a0dc5

**Comment by mieszko4** [comment]
Thnx @mattbretl for the comments! Please take a look again :)

**Review by mattbretl — APPROVED** [review]
LGTM

**Review by rune-developer — APPROVED** [review]
# Comprehensive Code Review: PR 24 ✅ APPROVED

## Overview
This PR successfully demonstrates `Rune.getChallengeNumber()` with two clear use patterns: seeded randomization and fixed list selection. The implementation is excellent example code.

## Assessment Summary

### ✅ Code Quality - Excellent
- Clear, self-documenting code with educational comments
- Well-structured demonstration of two distinct patterns
- Proper variable naming and code organization

### ✅ Security - No Issues
- Safe library usage (seedrandom - MIT licensed, well-maintained)
- No user input vulnerabilities
- No injection risks - all values internally generated
- Proper attribution and licensing compliance

### ✅ Performance - Optimal
- O(1) initialization with no runtime overhead
- Pre-calculated constants used in game loop
- Minimal memory footprint (~1.2KB for seedrandom)
- No conditional logic in hot path

### ✅ Maintainability - Excellent
- Self-documenting with clear comments explaining rationale
- Easy to extend with additional examples
- Version explicitly pinned (SDK 1.4.0)
- Local dependency management (no CDN risk)

## Key Strengths

1. **Educational Value**: Clearly demonstrates two different challenge number patterns
2. **Production Quality**: Clean, maintainable, follows best practices
3. **Proper Attribution**: Exemplary MIT license compliance for seedrandom
4. **Performance Conscious**: Map generation happens once, minimal overhead
5. **Clear Documentation**: Comments explain why, not just what

## Optional Enhancements (Low Priority)

1. Add brief comments for magic numbers:
   ```javascript
   const minSpeed = 0.005 // Slow enough to track visually
   const maxSpeed = 0.03  // Fast enough to be challenging
   ```

2. Document or remove commented line:
   ```javascript
   // const nextChallengeNumber = challengeNumber + 1;
   ```

## Verdict

**✅ APPROVED** - This is exemplary code that effectively teaches the challenge number API while maintaining production-quality standards. No blocking issues identified.

**Rating:** ⭐⭐⭐⭐⭐ Excellent example code

**Pull request merged**